sourcemash/api/categories.py

```diff
@@ -1,35 +1,55 @@
 from . import api
-from flask import abort
-from flask.ext.restful import Resource, fields, marshal
+from flask.ext.restful import Resource
+from flask.ext.security import current_user, login_required
 
 from sourcemash.models import Item
-from items import item_fields
 
 from sqlalchemy import func
 from collections import Counter
 
-category_fields = {
-    'category': fields.String,
-    'count': fields.String
-}
 
 class CategoryListAPI(Resource):
 
+    @login_required
     def get(self):
-        categories = Counter(dict(Item.query.with_entities(Item.category_1, func.count()).group_by(Item.category_1).all()))
-        categories.update(dict(Item.query.with_entities(Item.category_2, func.count()).group_by(Item.category_2).all()))
+        user_feed_ids = [feed.id for feed in current_user.subscribed]
+
+        categories = Counter()
+
+        distinct_category_1 = Item.query.with_entities(Item.category_1, func.count())   \
+                                        .filter(Item.feed_id.in_(user_feed_ids))        \
+                                        .group_by(Item.category_1)                      \
```

Comment:
If you don't do group by, you'll just get the {Category_1}, Count for each item in the list and categories will repeat. You need to do group_by to make sure each category only appears once and its count is the total of all appearances of the category.